File: C/URI/1074.c
#include <stdio.h>

int main(int argc, char const *argv[]) {
	int iterations, n, i;
	char even, positive;

	scanf("%d", &iterations);

	for (i = 0; i < iterations; ++i)
	{
		scanf("%d", &n);

		if(n == 0){
			printf("NULL\n");
			continue;
		}

		positive = (n > 0);

		even = (n % 2 == 0);

		printf("%s %s\n", (even) ? "EVEN":"ODD", (positive) ? "POSITIVE":"NEGATIVE");		
	}

	return 0;
}

File: Rust/URI/1074.rs
use std::io;
use std::num::ParseIntError;

fn read_int() -> Result<i32, ParseIntError> {
    let mut input_text = String::new();
    io::stdin()
        .read_line(&mut input_text)
        .expect("failed to read from stdin");

    let trimmed = input_text.trim();
    trimmed.parse::<i32>()
}

fn main() {
    let line_count = read_int().unwrap();

    for _ in 0..line_count {
        let current_number = read_int().unwrap();
        let mut current_result_str = String::new();

        if current_number == 0 {
            current_result_str.push_str("NULL");
        } else {
            if current_number%2 == 0 {
                current_result_str.push_str("EVEN")
            } else {
                current_result_str.push_str("ODD")
            }

            if current_number > 0 {
                current_result_str.push_str(" POSITIVE")
            } else {
                current_result_str.push_str(" NEGATIVE")
            }
        }

        println!("{}", current_result_str)
    }
}